op-node/rollup/derive/batch_mux.go

```diff
@@ -34,16 +34,21 @@ func NewBatchMux(lgr log.Logger, cfg *rollup.Config, prev NextBatchProvider, l2
 func (b *BatchMux) Reset(ctx context.Context, base eth.L1BlockRef, sysCfg eth.SystemConfig) error {
 	// TODO(12490): change to a switch over b.cfg.ActiveFork(base.Time)
 	switch {
-	default:
-		if _, ok := b.SingularBatchProvider.(*BatchQueue); !ok {
-			b.log.Info("BatchMux: activating pre-Holocene stage during reset", "origin", base)
-			b.SingularBatchProvider = NewBatchQueue(b.log, b.cfg, b.prev, b.l2)
+	case b.cfg.IsJovian(base.Time):
```

Comment:
Unsure if we need to modify the rollup derivation pipeline in general. If there's a better way let me know